plugins/modules/source_control/github/github_repo.py

```diff
@@ -0,0 +1,243 @@
+        changes['description'] = description
+
+    if changes:
+        if not check_mode:
+            repo.edit(**changes)
+
+        result['repo'].update({
+            'private': repo._private.value if not check_mode else private,
+            'description': repo._description.value if not check_mode else description
+        })
+        result['changed'] = True
+
+    return result
+
+
+def delete_repo(gh, name, organization=None, check_mode=False):
+    result = dict(changed=False)
+    if organization:
+        target = gh.get_organization(organization)
+    else:
+        target = gh.get_user()
+    try:
+        repo = target.get_repo(name=name)
+        if not check_mode:
+            repo.delete()
+        result['changed'] = True
+    except UnknownObjectException:
+        pass
+
+    return result
+
+
+def run_module(params, check_mode=False):
+    gh = authenticate(
+        username=params['username'], password=params['password'], access_token=params['access_token'])
+    if params['state'] == "absent":
+        return delete_repo(
+            gh=gh,
+            name=params['name'],
+            organization=params['organization'],
+            check_mode=check_mode
+        )
+    else:
+        return create_repo(
+            gh=gh,
+            name=params['name'],
+            organization=params['organization'],
+            private=params['private'],
+            description=params['description'],
+            check_mode=check_mode
+        )
+
+
+def main():
+    module_args = dict(
+        username=dict(type='str', required=False, default=None),
+        password=dict(type='str', required=False, default=None, no_log=True),
+        access_token=dict(type='str', required=False,
+                          default=None, no_log=True),
+        name=dict(type='str', required=True),
+        state=dict(type='str', required=False, default="present",
+                   choices=["present", "absent"]),
+        organization=dict(type='str', required=False, default=None),
+        private=dict(type='bool', required=False, default=False),
+        description=dict(type='str', required=False, default=''),
+    )
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True,
+        required_together=[('username', 'password')],
+        required_one_of=[('username', 'access_token')],
+        mutually_exclusive=[('username', 'access_token')]
+    )
+
+    if not HAS_GITHUB_PACKAGE:
+        module.fail_json(msg=missing_required_lib(
+            "PyGithub"), exception=GITHUB_IMP_ERR)
+
+    try:
+        result = dict(
+            changed=False
+        )
+        result = run_module(module.params, module.check_mode)
+        module.exit_json(**result)
+    except GithubException as e:
+        module.fail_json(msg="Github error. {0}".format(repr(e)), **result)
+    except Exception as e:
+        module.fail_json(msg="Unexpected error. {0}".format(repr(e)), **result)
```

Comment:
`result` only contains `changed=False` (which is the default value for `changed`), so it's not needed here:
```suggestion
        module.fail_json(msg="Github error. {0}".format(repr(e)))
    except Exception as e:
        module.fail_json(msg="Unexpected error. {0}".format(repr(e)))
```